pkg/pool-stable/test/foundry/E2eSwap.t.sol

```diff
@@ -36,13 +38,46 @@ contract E2eSwapStableTest is E2eSwapTest {
         maxPoolSwapFeePercentage = 10e16;
     }
 
-    function calculateMinAndMaxSwapAmounts() internal override {
-        minSwapAmountTokenA = poolInitAmountTokenA / 1e3;
-        minSwapAmountTokenB = poolInitAmountTokenB / 1e3;
+    function calculateMinAndMaxSwapAmounts() internal virtual override {
+        uint256 rateTokenA = getRate(tokenA);
+        uint256 rateTokenB = getRate(tokenB);
+
+        // The vault does not allow trade amounts (amountGivenScaled18 or amountCalculatedScaled18) to be less than
+        // MIN_TRADE_AMOUNT. For "linear pools" (PoolMock), amountGivenScaled18 and amountCalculatedScaled18 are
+        // the same. So, minAmountGivenScaled18 > MIN_TRADE_AMOUNT. To derive the formula below, note that
+        // `amountGivenRaw = amountGivenScaled18/(rateToken * scalingFactor)`. There's an adjustment for stable math
+        // in the following steps.
+        uint256 tokenAMinTradeAmount = MIN_TRADE_AMOUNT.divUp(rateTokenA).mulUp(10 ** decimalsTokenA);
+        uint256 tokenBMinTradeAmount = MIN_TRADE_AMOUNT.divUp(rateTokenB).mulUp(10 ** decimalsTokenB);
```

Comment:
If we removed min trade amount for the purpose of the tests, we could just use 0 right?

A bit late to the party, but see #929. This and the batch swap tests made me rethink this constraint a bit; the Vault should still work without it to the best of our knowledge.

This PR can continue with this, but then we should review do / undo tests so that they also work with very low amounts.

Reply:
Nice, it makes sense! I'd still have a min trade amount, to avoid an `amountGiven = 0` in the second swap, but I agree that removing MIN_TRADE_AMOUNT makes the life easier.